### DISCUSSION TITLE
problems with the hourly simulation

### DISCUSSION THREAD

**Discussion by fullflowwhere** [mhm-ufz/mHM#110] [Q&A]
I have just started working with the MHM model. Currently, I'm attempting to run the model on an hourly scale instead of the daily scale to obtain results. I have converted the data of "pre", "pet", "tavg", and "tann" in "meteo" within the "test_domain" to the hourly scale. Also, I have increased the amount of data in "gauge" to match the same number of time steps as in "meteo". However, when running the MHM model, the system still encounters errors. Could you please tell me how to solve this problem?
[input.zip](https://github.com/user-attachments/files/17828347/input.zip)
![faa438f868b30eba405803e2f589e04](https://github.com/user-attachments/assets/083b25ca-e4a5-4fd6-9c75-ff02063e137c)

**Comment by pallavhydro** [comment]
Hi @fullflowwhere 

Welcome to the mHM community!

I checked your input files. Following are my observations:

1. Your editing of gauge time series `00398.txt` to hourly format looks fine.
2. Your editing of `pre.nc` to hourly format has issues. 
3. The reason mHM is throwing the error while reading `pre.nc` is that all the time stamps are identical (screenshot 1). They need to be hourly propagating. Additionally, there are multiple levels generated for the variable pre, which should not be the case. Please refer to the original `pre.nc`. 
4. I also observe that the structure of the `pre.nc` that you edited is quite different to the original (screenshot 2). The best practice would be to keep to the original structure and only make structural changes that are required e.g., `time:units = "hours since ...`
5. I didn't check other meteo files but please take comments 3 and 4 for them as well.

### mHM at hourly resolution
I have further comments on using mHM for hourly simulations
6. mHM always runs at hourly scale, internally, irrespective of model input.
7. The temporal resolution of the netCDF spatiotemporal output `mhm_fluxes_states.nc` and `mrm_fluxes_states.nc` can be controlled using `timeStep_model_outputs` field in the `mhm_outputs.nml` and `mhm_outputs.nml`, respectively.
8. The temporal resolution of point time series output i.e. the `discharge.nc` and `daily_discharge.out` is controlled by the resolution of the input gauge observations (in your case `00398.txt`).
9. Which means, one can retrieve hourly (sub daily) point time series discharge output from mHM with `daily meteo` files as long as `hourly input gauge observation/s` is provided. While hourly (sub-daily) meteo forcings would improve hourly discharge characteristics `hourly forcings are not required for hourly output`.
10. Which also means, hourly meteo and daily input gauge observations will output daily point discharge output from mHM. 

I hope this helps you go past the current error and/or improve your modeling strategy.

Pallav

Screenshot 1
![Screenshot 2024-11-20 at 11 50 28](https://github.com/user-attachments/assets/ce0425f7-1ee9-4d80-8df0-83ebe84d290f)

Screenshot 2
![Screenshot 2024-11-20 at 12 01 01](https://github.com/user-attachments/assets/80fe3f7d-2efc-421a-b81a-cb7b44781dcd)

**Comment by fullflowwhere** [comment]
Thank you for your reply. I have modified the nc file format in meteo to be similar to that of the source file according to your instructions. However, when I was running MHM, the following error occurred again: ***ERROR: read_forcing_nc: expected daily input forcing, but read something else. Ensure all input time steps have the same units across all input files. So I also changed the data in the meteo of domain2 to an hourly scale. But now the following error pops up: ERROR: wanted to read 35064 time-steps from test_domain_2/input/gauge/45.txt, but reached end-of-file at 35064. Note: The following floating-point exceptions are signalling: IEEE_INVALID_FLAG. I have been trying to modify the data for a long time but still haven't managed to fix it. I have no choice but to turn to you for help.






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "mhm-ufz/mHM"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年11月20日(星期三) 晚上7:36
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [mhm-ufz/mHM] problems with the hourly simulation (Discussion #110)





 
Hi @fullflowwhere
 
Welcome to the mHM community!
 
I checked your input files. Following are my observations:
  
Your editing of gauge time series 00398.txt to hourly format looks fine.
 
Your editing of pre.nc to hourly format has issues.
 
The reason mHM is throwing the error while reading pre.nc is that all the time stamps are identical (screenshot 1). They need to be hourly propagating. Additionally, there are multiple levels generated for the variable pre, which should not be the case. Please refer to the original pre.nc.
 
I also observe that the structure of the pre.nc that you edited is quite different to the original (screenshot 2). The best practice would be to keep to the original structure and only make structural changes that are required e.g., time:units = "hours since ...
 
I didn't check other meteo files but please take comments 3 and 4 for them as well.
  
mHM at hourly resolution
 
I have further comments on using mHM for hourly simulations
 6. mHM always runs at hourly scale, internally, irrespective of model input.
 7. The temporal resolution of the netCDF spatiotemporal output mhm_fluxes_states.nc and mrm_fluxes_states.nc can be controlled using timeStep_model_outputs field in the mhm_outputs.nml and mhm_outputs.nml, respectively.
 8. The temporal resolution of point time series output i.e. the discharge.nc and daily_discharge.out is controlled by the resolution of the input gauge observations (in your case 00398.txt).
 9. Which means, one can retrieve hourly (sub daily) point time series discharge output from mHM with daily meteo files as long as hourly input gauge observation/s is provided. While hourly (sub-daily) meteo forcings would improve hourly discharge characteristics hourly forcings are not required for hourly output.
 10. Which also means, hourly meteo and daily input gauge observations will output daily point discharge output from mHM.
 
I hope this helps you go past the current error and/or improve your modeling strategy.
 
Pallav
 
Screenshot 1
 Screenshot.2024-11-20.at.11.50.28.png (view on web)
 
Screenshot 2
 Screenshot.2024-11-20.at.12.01.01.png (view on web)

**Comment by pallavhydro** [comment]
Hi @fullflowwhere 

Please share the files like the last time you did or just screenshots of `cdo sinfon` and `ncdump -h` on the meteo files. Without that it's hard to speculate the source of the current problem.

Pallav

**Comment by fullflowwhere** [comment]
Hello. Could you please tell me how to solve this error? This problem has been bothering me for over a week. I've checked the data several times, but still haven't found the problem. 
I'm eager to get your help and would be extremely grateful!


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "mhm-ufz/mHM"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年11月25日(星期一) 晚上6:47
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [mhm-ufz/mHM] problems with the hourly simulation (Discussion #110)





 
Hi @fullflowwhere
 
Please share the files like the last time you did or just screenshots of cdo sinfon and ncdump -h on the meteo files. Without that it's hard to speculate the source of the current problem.
 
Pallav

**Comment by pallavhydro** [comment]
Hi @fullflowwhere 

Please share the following as a zip file so that we could solve the problem:

- mhm.nml
- All meteo files (pre.nc, ...)
- All gauge files (45.txt, ...)

Pallav

**Comment by fullflowwhere** [comment]
This is a screenshot of the meteo data in domain2.
Thank you for your help.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "mhm-ufz/mHM"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年12月6日(星期五) 晚上8:40
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [mhm-ufz/mHM] problems with the hourly simulation (Discussion #110)





 
Hi @fullflowwhere
 
Please share the following as a zip file so that we could solve the problem:
  
mhm.nml
 
All meteo files (pre.nc, ...)
 
All gauge files (45.txt, ...)
  
Pallav

**Comment by pallavhydro** [comment]
Hi

I do not find any screenshot on your last message.

Also, it is 45.txt that is prompted to have an error based on your previous message/s. Would you share the file/ screenshot of the file content as well please.

P

**Comment by fullflowwhere** [comment]
Thank you very much for your patient reply.
I noticed that this email contains a screenshot compressed package. There might be some errors. 
This compressed package includes screenshots of the parameters in "meteo" as well as the "45.txt" file. 
Thank you very much for your help. 
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "mhm-ufz/mHM"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年12月6日(星期五) 晚上8:56
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [mhm-ufz/mHM] problems with the hourly simulation (Discussion #110)





 
Hi
 
I do not find any screenshot on your last message.
 
Also, it is 45.txt that is prompted to have an error based on your previous message/s. Would you share the file/ screenshot of the file content as well please.
 
P
